R/forecast-bbs-core.R

```diff
@@ -72,8 +72,13 @@ combine_subspecies = function(df){
   }
 
   df %>%
-    group_by(site_id, year, species_id, lat, long) %>%
+    group_by(site_id, year, species_id) %>%
     summarize(abundance = sum(abundance)) %>%
+    ungroup() %>%
+    left_join(
+      distinct(dplyr::select(ungroup(df), -abundance, -species_id)),
+      by = c("site_id", "year")
+    ) %>%
```

Comment:
Can you walk me through this set of changes starting on line 74. The previous version made sense to me (after updating the names re-sum the abundances to deal with cases where multiple sub-species were present at a site), but I don't fully follow what's going on now. Maybe also add a comment at the start of this pipe to explain what is happening.